src/classes/AuthenticationSession.ts

```diff
@@ -178,4 +181,18 @@ export class AuthenticationSession {
       this.authContext.reject();
     });
   }
+
+  private obtainNewAuthTokenUsingRefreshToken(): void {
```

Comment:
I know this isn't yet being called anywhere, but if I may suggest, you probably want a method like `getAccessToken` that examines the expiration date - which you'll also need to track - and will transparently refresh if the access token is too old.

Ideally, this would be provided by your client library, but unfortunately this is still an open issue upstream: https://github.com/FusionAuth/fusionauth-typescript-client/issues/74 .

I haven't looked closely enough to say if this is a worthwhile tradeoff, but as an alternative to implementing all of this security code yourself, you may want to consider switching to another client library. Unless you're doing anything FusionAuth-specific, you should hopefully be able to use any OAuth/OIDC client library.

If you do decide to switch clients, I'd suggest doing so as its own stand-alone PR, before implementing this additional functionality.